common/trinary/trit_array.c

```diff
@@ -26,38 +26,44 @@ size_t flex_trit_array_slice(const flex_trit_t *trit_array,
   uint8_t tshift = (start % 4U) << 1U;
   uint8_t rshift = (8U - tshift) % 8U;
   size_t index = start >> 2U;
-  size_t max_index = (start + num_trits - 1) >> 2U;
+  size_t end_index = (start + num_trits - 1) >> 2U;
   size_t i, j;
   // Calculate the number of bytes to copy over
   for (i = index, j = 0; i < index + num_bytes; i++, j++) {
     buffer = trit_array[i];
     buffer = buffer >> tshift;
-    if (rshift && i < max_index) {
-      buffer |= (trit_array[i + 1] << rshift);
+    if (i < end_index) {
```

Comment:
You could do something like:
byte_t buffer = 0;
int trits_count = 0;
for (i = start; i < start + num_trits; ++i){
uint8_t shift = (trits_count % 4)<<1U
b |= (flex_trit_array_at(i)<<shift);
if ((++trits_count % 4) == 0){
buffer = 00X0;
to_trit_array[trits_count/4 - 1] = buffer;
}
and that way you can utilize `flex_trit_array_at` and maybe share the same code with FIVE_TRITS_PER_BYTE
didn't try it though ..

Reply:
though current version is 4 time faster, forget about it ;)